db/models.py

```diff
@@ -1,52 +1,138 @@
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager, User
 from django.db import models
+from django.core.exceptions import ValidationError
+from django.contrib.auth import get_user_model
 
 
-class Genre(models.Model):
-    name = models.CharField(max_length=255, unique=True)
+class CustomUserManager(BaseUserManager):
+    def create_user(self,
+                    username: str,
+                    password: str,
+                    **extra_fields) -> User:
 
-    def __str__(self) -> str:
-        return self.name
+        if not username:
+            raise ValueError("The Username field must be set")
+        user = self.model(username=username, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
 
+    def create_superuser(self,
+                         username: str,
+                         password: str,
+                         **extra_fields) -> User:
```

Comment:
Similar to the `create_user` method, the return type hint `-> User` should be `-> 'User'` or `-> models.Model` to prevent circular import issues.